services/web/server/src/simcore_service_webserver/meta_projects.py

```diff
@@ -0,0 +1,83 @@
+""" Access to the to projects module
+
+    - Adds a middleware to intercept /projects/* requests
+    - Implements a MetaProjectRunPolicy policy (see director_v2_abc.py) to define how meta-projects run
+
+"""
+
+
+import logging
+import re
+from typing import List, Tuple
+
+from aiohttp import web
+from aiohttp.web_middlewares import _Handler
+from models_library.basic_regex import UUID_RE
+from models_library.projects import ProjectID
+
+from ._meta import api_version_prefix as VTAG
+from .director_v2_api import AbstractProjectRunPolicy
+from .meta_iterations import get_or_create_runnable_projects, get_runnable_projects_ids
+from .meta_version_control import CommitID, VersionControlForMetaModeling
+from .projects.projects_handlers import RQ_REQUESTED_REPO_PROJECT_UUID_KEY
+
+log = logging.getLogger(__name__)
+
+
+# SEE https://github.com/ITISFoundation/osparc-simcore/blob/master/services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml#L8563
+URL_PATTERN = re.compile(rf"^\/{VTAG}\/projects\/({UUID_RE})[\/]{{0,1}}")
+
+
+@web.middleware
+async def projects_redirection_middleware(request: web.Request, handler: _Handler):
+    """Intercepts /projects/{project_id}* requests and redirect them to the copy @HEAD"""
+
+    if URL_PATTERN.match(f"{request.rel_url}"):
+        #
+        # FIXME: because hierarchical design is not guaranteed, we find ourselves with
+        # entries like /v0/computation/pipeline/{project_id}:start which might also neeed
+        # indirection
```

Comment:
what is the consequence of that? is it still the case?

Reply:
Here I point out a defect in the design of the API that could be overcome if we would follow a hierarchical approach (which i propose for the new projects API). Using https://google.aip.dev/121, the project resource should either incorporate a ``run``  or have a sub-resource (e.g. ``workbench`` or ``nodes``) that has it.

I would propose something like
```
Create:   POST    /project
List:        GET       /projects
Get:        GET       /projects/{project_uuid}
Update:  PATCH  /projects/{project_uuid}
Replace  PUT      /projects/{project_uuid}

Open        POST   /projects/{project_uuid}:open
Close        POST   /projects/{project_uuid}:close

Start        POST   /projects/{project_uuid}:start
Stop        POST   /projects/{project_uuid}:stop
```
And some extra sub-collections like ``/projects/{project_uuid}/nodes``